src/quo2/components/dropdowns/old_button_style.cljs

```diff
@@ -0,0 +1,266 @@
+                      :label-color          colors/white
+                      :background-color     {:default  colors/neutral-80-opa-40
+                                             :pressed  colors/neutral-80-opa-50
+                                             :disabled colors/neutral-80-opa-40}}
+    :blur-bg         {:icon-color           colors/white
+                      :icon-secondary-color colors/white-opa-70
+                      :label-color          colors/white
+                      :background-color     {:default  colors/white-opa-5
+                                             :pressed  colors/white-opa-10
+                                             :disabled colors/white-opa-5}}
+    :blurred         {:icon-color            colors/white
+                      :icon-secondary-color  colors/white
+                      :icon-background-color {:default colors/neutral-80
+                                              :blurred colors/white-opa-10}
+                      :label-color           colors/white
+                      :background-color      {:default  colors/neutral-90
+                                              :pressed  colors/neutral-90
+                                              :disabled colors/neutral-90-opa-10-blur}
+                      :blur-overlay-color    colors/neutral-80-opa-40
+                      :blur-type             :dark}
+    :blur-bg-outline {:icon-color           colors/white
+                      :icon-secondary-color colors/white-opa-40
+                      :label-color          colors/white
+                      :border-color         {:default  colors/white-opa-10
+                                             :pressed  colors/white-opa-20
+                                             :disabled colors/white-opa-5}}
+    :shell           {:icon-color       colors/white
+                      :label-color      colors/white
+                      :background-color {:default colors/neutral-95}}}})
+
+(defn shape-style-container
+  [type icon size]
+  {:height        size
+   :border-radius (if (and icon (#{:primary :secondary :danger} type))
+                    24
+                    (case size
+                      56 12
+                      40 12
+                      32 10
+                      24 8))})
+
+(defn style-container
+  [{:keys [type size disabled background-color border-color icon above width before after blur-active?]}]
+  (merge {:height             size
+          :align-items        :center
+          :justify-content    :center
+          :flex-direction     (if above :column :row)
+          :padding-horizontal (when-not (or icon before after)
+                                (case size
+                                  56 16
+                                  40 16
+                                  32 12
+                                  24 8))
+          :padding-left       (when-not (or icon before)
+                                (case size
```

Comment:
It shouldn't happen, but a missing `case` condition will throw an error to the user.

Reply:
size has a default value in this component so it should be alright here 👌

Reply:
@J-Son89 What I mean is that when using `case` it's safer to have a default condition if no value matches. The UI code can usually deal with nils, but not with exceptions.

```clojure
(let [size true]
  (case size
    56 12
    40 12
    32 10
    24 8))
;; throws "No matching clause: true"
```

So from the perspective of `style-container`, `size` can be anything, and if it's not one of 56, 40, 32, or 24 it throws. The fact that a component has a default value is unknown to this function. This coupling between functions I find unreliable as a practice.

```clojure
(style-container {:size 42})
;; throws "No matching clause: true"
```

If this was Clojure JVM I wouldn't be so worried, but exceptions are not handled gracefully in the mobile app. Here's what a user sees when a `production` build has a no matching clause exception. The app explodes and if they press `Cancel` they're greeted with an empty white screen and have to close the app and start all over.

<img src="https://github.com/status-im/status-mobile/assets/46027/8bf2f00f-a0b0-4294-9e22-81c49c8ddb31" height="500" />

----

In my opinion, we should basically not take this risk in this RN app and I wish clj-kondo had a linter to fail if no default condition is provided.

I'm cc'ing @flexsurfer and @ulisesmac to share this knowledge too.

Reply:
btw, is there no alternative error screen we can use? React has this https://react.dev/reference/react/Component#catching-rendering-errors-with-an-error-boundary but not sure what's possible for React Native

Reply:
I don't know either because I only used error boundaries sporadically in non-reagent apps, but I do like the idea @J-Son89. It's totally worth having an issue about this problem if we don't already have one.

In the repo we can find this comment:

> Downgrade reagent error to warning Reagent uses console.error to notify the user
that an exception occurs while rendering. Unfortunately, react-native can only
show one RedBox at a time and discards all subsequent ones. This obscures the
actual exception. By downgrading the first screen to a warning, the second
screen is actually shown to the user.
>
> -- namespace status-im2.setup.global-error

So I think we would need custom code to make things work well with error boundaries.

Reply:
There is the package [`react-native-exception-handler`](https://www.npmjs.com/package/react-native-exception-handler) which will help us catch both JS and native (Java/Obj-C) exceptions.

Reply:
Yes, I'm aware of this risk when using `case`, so I always add that default clause when using it.

It is so weird that ClojureScript preferred throwing and exception instead of just `nil` since this is functional programming. I also was thinking of redefining `case` for a safer use. 🤔

I will warn other devs in code reviews about this when I find it 👍